backend/app/api/routes/podcasts.py

```diff
@@ -0,0 +1,151 @@
+from sqlmodel import select
+from sqlalchemy.orm import selectinload
+from app.core.config import settings
+from app.models.podcast import Podcast
+from app.schemas.public import PodcastPublic, PodcastsPublic
+from app.services.podcast_service import generate_podcast_for_course
+
+router = APIRouter(prefix="/podcasts", tags=["podcasts"])
+
+
+@router.get("/{course_id}", response_model=PodcastsPublic)
+def list_podcasts(course_id: uuid.UUID, session: SessionDep, current_user: CurrentUser) -> Any:
+    pods = session.exec(select(Podcast).where(Podcast.course_id == course_id)).all()
+    return PodcastsPublic(data=[PodcastPublic.model_validate(p) for p in pods])
+
+
+class GeneratePodcastRequest(BaseModel):
+    title: Optional[str] = None
+    mode: Optional[str] = None  # 'dialogue' | 'presentation'
+    topics: Optional[str] = None
+    teacher_voice: Optional[str] = None
+    student_voice: Optional[str] = None
+    narrator_voice: Optional[str] = None
+    document_ids: Optional[List[uuid.UUID]] = None
+
+
+@router.post("/{course_id}/generate", response_model=PodcastPublic)
+async def generate_podcast(
+    course_id: uuid.UUID,
+    session: SessionDep,
+    current_user: CurrentUser,
+    body: GeneratePodcastRequest | None = None,
+) -> Any:
+    if not body or not body.title or not body.title.strip():
+        raise HTTPException(status_code=422, detail="Title is required")
+    title = body.title.strip()
+    mode = (body.mode or "dialogue") if body else "dialogue"
+    topics = body.topics if body else None
+    teacher_voice = body.teacher_voice if body and body.teacher_voice else settings.PODCAST_TEACHER_VOICE
+    student_voice = body.student_voice if body and body.student_voice else settings.PODCAST_STUDENT_VOICE
+    narrator_voice = body.narrator_voice if body and body.narrator_voice else settings.PODCAST_TEACHER_VOICE
+    doc_ids = body.document_ids if body and body.document_ids else None
+    podcast = await generate_podcast_for_course(
+        session,
+        course_id,
+        title,
+        teacher_voice,
+        student_voice,
+        narrator_voice,
+        mode,
+        topics,
+        doc_ids,
+    )
+    return PodcastPublic.model_validate(podcast)
+
+
+@router.get("/by-id/{podcast_id}", response_model=PodcastPublic)
+def get_podcast(podcast_id: uuid.UUID, session: SessionDep, current_user: CurrentUser) -> Any:
+    pod = session.get(Podcast, podcast_id)
+    if not pod:
+        raise HTTPException(status_code=404, detail="Podcast not found")
+    return PodcastPublic.model_validate(pod)
+
+
+@router.get("/by-id/{podcast_id}/audio")
+def stream_audio(podcast_id: uuid.UUID, session: SessionDep, current_user: CurrentUser):
+    pod = session.get(Podcast, podcast_id)
+    if not pod:
+        raise HTTPException(status_code=404, detail="Podcast not found")
+    if pod.storage_backend == "local":
+        file_path = pod.audio_path
+        if not os.path.exists(file_path):
```

Comment:
I presume the file is on the server at this point. Where does it get deleted?

Reply:
In theory, once the podcast is deleted, the local file should be deleted. It is deleted [here](https://github.com/A3copilotprogram/plg_6/blob/0bbe90753d5539597914c0e716767c665dfcf57c/backend/app/api/routes/podcasts.py#L118).

However, this is just a defensive check to safely read the file by checking if it exists, since we are dealing with file paths, considering it is possible to have a reference file path remaining in the DB while the actual local file has been deleted. More likely to happen locally, e.g, if you clean up the docker backend build but db records still persist.

_Also this whole local option is just for easy testing without everyone needing to setup AWS S3 since we don't use shared credentials due to cost_